models/vrp.rb

```diff
@@ -477,6 +478,20 @@ def self.generate_schedule_indices_from_date(hash)
       hash
     end
 
+    def self.ensure_timewindows_start_is_zero(hash)
+      [hash[:services], hash[:shipments]].compact.flatten.each{ |s|
+        [s[:activity], s[:activities], s[:pickup], s[:delivery]].compact.flatten.each{ |a|
+          a[:timewindows]&.each{ |tw| tw[:start] ||= 0 }
+        }
+      }
+
+      hash[:vehicles]&.each{ |v|
+        [v[:timewindow], v[:sequence_timewindows]].compact.flatten.each{ |tw|
+          tw[:start] ||= 0
+        }
+      }
+    end
+
```

Comment:
So the default start value of `model::timewindow` is not enough, is that it? :thinking: I am confused. When I asked "we need to make sure..." I wasn't expecting that we need such a function. Is the source of the problem, we let the users to send {"start": null} ?

Reply:
If that is the case I wonder if "allow_blank: false"  would  be a better solution :nerd_face:  lets talk about it in the daily tomorrow

Reply:
Right, this might be better ! I did not think about it.

Reply:
However, this might return an error for VRP that were previously accepted. As you said, let's talk about this during daily :)

Reply:
We can't add allow_blank: false. The timewindows transmitted through a CSV without a start will always be in error

Reply:
Before I forget all, we said it would be better to do all of this by modifying the `coerce_with`. It  will have an input argument `default=nil` to be used when the parameter is `nil`. It will be used for the parameter for which we don't want to use the default value logic of the grape entity directly and we cannot do `allow_blank: false`. Feel free to add if I missed any details.

Reply:
In fact it does not seem possible to use coerce_with to define default values (https://github.com/ruby-grape/grape/pull/2040#issuecomment-615483919).

Either we accept that users are aware default start value is 0 and we can easily define `default: 0` within grape, or we do need this function.